docs/ref/csp.txt

```diff
@@ -0,0 +1,260 @@
+.. _content-security-policy:
+
+=======================
+Content Security Policy
+=======================
+
+.. versionadded:: 6.0
+
+.. module:: django.middleware.csp
+   :synopsis: Middleware for Content Security Policy headers
+
+Content Security Policy (CSP) is a web security standard that helps prevent
+content injection attacks by restricting the sources from which content can be
+loaded. It plays an important role in a comprehensive :ref:`security strategy
+<security-csp>`.
+
+For configuration instructions in a Django project, see the :ref:`Using CSP
+<using-csp>` documentation. For an HTTP guide about CSP, see the `MDN Guide on
+CSP <https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/CSP>`_.
+
+Overview
+========
+
+The `Content-Security-Policy specification <https://www.w3.org/TR/CSP3/>`_
+defines two complementary headers:
+
+* ``Content-Security-Policy``: Enforces the CSP policy, blocking content that
+  violates the defined directives.
+* ``Content-Security-Policy-Report-Only``: Reports CSP violations without
+  blocking content, allowing for non-intrusive testing.
+
+Each policy is composed of one or more directives and their values, which
+together instruct the browser on how to handle specific types of content.
+
+When the :class:`~django.middleware.csp.ContentSecurityPolicyMiddleware` is
+enabled, Django automatically builds and attaches the appropriate headers to
+each response based on the configured :ref:`settings <csp-settings>`, unless
+they have already been set by another layer.
+
+.. _csp-settings:
+
+Settings
+========
+
+The :class:`~django.middleware.csp.ContentSecurityPolicyMiddleware` is
+configured using the ``SECURE_CSP`` and ``SECURE_CSP_REPORT_ONLY`` settings.
+
+:setting:`SECURE_CSP`
+
+Defines the **enforced Content Security Policy**.
+
+When set, the middleware adds a ``Content-Security-Policy`` HTTP header to
+responses. This instructs the browser to actively block any content that
+violates the configured directives.
+
+.. _content-security-policy-report-only:
+
+:setting:`SECURE_CSP_REPORT_ONLY`
+
+Defines a **report-only Content Security Policy**.
+
+When set, the middleware adds a ``Content-Security-Policy-Report-Only`` HTTP
+header to responses. This allows the browser to monitor and report policy
+violations without blocking content, making it useful for refining a policy.
+
+Most browsers log CSP violations to the developer console and can optionally
+send them to a reporting endpoint. To collect these reports, the ``report-uri``
+directive must be defined (see :ref:`csp-reports` for more details).
```

Comment:
Note that we have separate docs for these settings. To reduce duplication, we can use the csrf docs as a template here:
```suggestion
The following settings can be used to control Django's CSP behavior:

* :setting:`SECURE_CSP`
* :setting:`SECURE_CSP_REPORT_ONLY`
```
Then any information that is only included here could be included in the settings docs